### ISSUE TITLE
Upgrade to JRuby 1.7.26

### ISSUE THREAD

**Issue by gunnarmorling** [asciidoctor/asciidoctor-maven-plugin#276]
Could an update to the latest JRuby 1.7.x be done? This let's us use this plug-in on Java 9 (despite the comments on https://github.com/asciidoctor/asciidoctorj/issues/515 stating that JRuby 9.x would be needed).

I reckon there may be issues left which are only fixed in JRuby 9.x, but doing the update to the latest micro seems like a reasonable step nevertheless.

**Comment by abelsromero** [comment]
If you need to upgrade the dependency I guess you know you can do it yourself in the pom ([example](https://github.com/asciidoctor/asciidoctor-maven-examples/blob/master/asciidoc-to-html-example/pom.xml#L25-L38)). So, this should not be a stopper for you.
Regarding the actual upgrade, I see it likely to happen, but not soon. We align with Asciidoctorj, so as soon as v1.5.5 is released we will align to it, test and release too.

If you want, you can open an issue on Asciidoctorj to discuss it, or send a PR if all tests pass with that version.

**Comment by gunnarmorling** [comment]
> you can do it yourself in the pom

Yes, I know (that's what we are doing currently).

I hoped I could quickly update it here in the plug-in itself, but it seems more involved than I'd have expected from such a micro version update :( I also tried to go to 1.7.25 and ADJ 1.5.4.1 but hit some errors there, too (just seeing now that there is another pending PR for doing exactly that). My knowledge of AsciiDoctorJ and this plug-in is essentially zero, so I fear I won't be a big help. Sorry, I thought it'd be a quick win :/

The update to 1.7.26 on our side makes things work it seems, but these errors above make me a bit concerned.

**Comment by abelsromero** [comment]
Don't worry, we welcome every collaboration :+1: 

> I hoped I could quickly update it here in the plug-in itself, but it seems more involved than I'd have expected from such a micro version update :( I also tried to go to 1.7.25 and ADJ 1.5.4.1 but hit some errors there

Truth is that there's no perfect match, some extensions like pdf may require higher versions of jRuby, but then other don't. It's a tough balance.
Truth is that Maven and Gradle plugins try to makes things easier with default versions. But in AsciidoctorJ, jRuby is a 'provided' dependency, so it forces you to be aware of jRuby and choose what version you need for your project.

I'll close this issue if everything is clear. Feel free to ask anything you need.

**Comment by Sanne** [comment]
FYI I'm running the plugin by overriding the JRuby version to `9.1.14.0` to get it to work on JDK9.

Would be great to apply this update in the plugin?

**Comment by abelsromero** [comment]
I trully don't have a strong opinion on this topic, but I'll re-iterate on my previous comment.
TL;DR; we align with minimal version of AsciidoctorJ.

What we could do is upgrade de note about versionsing in the README to include minimal version for PDF, Java9 or other cases.

**Comment by slachiewicz** [comment]
Small update - based on the release note for AsciidoctorJ 1.5.6:
> Upgrade JRuby to 1.7.26 and 9.1.12.0 (The distribution still uses 9.1.8.0 though for Windows compatibility)

**Comment by abelsromero** [comment]
Not relevant anymore, current jruby is 9.2.11.1 and tests include java 9, 10, and 11.
